core/metadata.js

```diff
@@ -217,37 +218,49 @@ function isUpToDate (side /*: SideName */, doc /*: Metadata */) {
 // Return true if the metadata of the two folders are the same
 // For updated_at, we accept up to 3s of differences because we can't
 // rely on file systems to be precise to the millisecond.
+
+const makeDiffFilter = (interestingFields) => (path, key) => {
+  return path.length === 0 && interestingFields.indexOf(key) === -1
+}
+
+const sameFolderFilter = makeDiffFilter([
+  '_id', 'docType', 'remote', 'tags', 'trashed', 'ino'])
+
 function sameFolder (one /*: Metadata */, two /*: Metadata */) {
   const {path} = two
-  let fields = ['_id', 'docType', 'remote', 'tags', 'trashed', 'ino']
-  one = pick(one, fields)
-  two = pick(two, fields)
-  const same = isEqual(one, two)
-  if (!same) log.trace({path, diff: {one, two}})
-  return same
+  let diff = delta(one, two, sameFolderFilter)
```

Comment:
Why not just keep `one` and `two` and `deepDiff()` those? Wouldn't this prevent the need for the various filters?

Or even simpler:

```js
function partialDiff(one, two, keys) {
  one = pick(one, keys)
  two = pick(two, keys)
  return deepDiff(one, two)
}
```

Wouldn't this work?

Reply:
Might be premature optimization, but this is much more efficient and as readable.